src/crypto_utils.ts

```diff
@@ -1,118 +1,143 @@
-import crypto from "crypto"
+import forge from 'node-forge'
 import {BaseWallet, ethers, getBytes, SigningKey, solidityPackedKeccak256} from "ethers"
 import * as fs from "node:fs";
 
-const block_size = 16 // AES block size in bytes
-const hexBase = 16
+const BLOCK_SIZE = 16 // AES block size in bytes
+const HEX_BASE = 16
 
-export function encrypt(key: Buffer, plaintext: Buffer): { ciphertext: Buffer; r: Buffer } {
+export function encrypt(key: Uint8Array, plaintext: Uint8Array): { ciphertext: Uint8Array; r: Uint8Array } {
     // Ensure plaintext is smaller than 128 bits (16 bytes)
-    if (plaintext.length > block_size) {
+    if (plaintext.length > BLOCK_SIZE) {
         throw new RangeError("Plaintext size must be 128 bits or smaller.")
     }
+
     // Ensure key size is 128 bits (16 bytes)
-    if (key.length != block_size) {
+    if (key.length != BLOCK_SIZE) {
         throw new RangeError("Key size must be 128 bits.")
     }
-    // Create a new AES cipher using the provided key
-    const cipher = crypto.createCipheriv("aes-128-ecb", key, null)
-
     // Generate a random value 'r' of the same length as the block size
-    const r = crypto.randomBytes(block_size)
+    const r = forge.random.getBytesSync(BLOCK_SIZE)
+
+    // Create a new AES cipher using the provided key
+    const cipher = forge.cipher.createCipher('AES-ECB', forge.util.createBuffer(key))
 
     // Encrypt the random value 'r' using AES in ECB mode
-    const encryptedR = cipher.update(r)
+    cipher.start()
+    cipher.update(forge.util.createBuffer(r))
+    cipher.finish()
+
+    // Get the encrypted random value 'r' as a Buffer and ensure it's exactly 16 bytes
+    // the node-forge package produces strange results when passing strings
+    const encryptedR = encodeString(cipher.output.data).slice(0, BLOCK_SIZE)
 
     // Pad the plaintext with zeros if it's smaller than the block size
-    const plaintext_padded = Buffer.concat([Buffer.alloc(block_size - plaintext.length), plaintext])
+    const plaintextPadded = new Uint8Array([...new Uint8Array(BLOCK_SIZE - plaintext.length), ...plaintext])
 
     // XOR the encrypted random value 'r' with the plaintext to obtain the ciphertext
-    const ciphertext = Buffer.alloc(encryptedR.length)
-    for (let i = 0; i < encryptedR.length; i++) {
-        ciphertext[i] = encryptedR[i] ^ plaintext_padded[i]
+    const ciphertext = new Uint8Array(BLOCK_SIZE)
+
+    for (let i = 0; i < BLOCK_SIZE; i++) {
+        ciphertext[i] = encryptedR[i] ^ plaintextPadded[i]
     }
 
-    return {ciphertext, r}
+    return {
+        ciphertext,
+        r: encodeString(r)
+    }
 }
 
-export function decrypt(key: Buffer, r: Buffer, ciphertext: Buffer): Buffer {
-    if (ciphertext.length !== block_size) {
+export function decrypt(key: Uint8Array, r: Uint8Array, ciphertext: Uint8Array): Uint8Array {
```

Comment:
some of the code in decrypt and encrypt is the same, like validaty checks - why not to combine to a single method?